torchaudio/functional/functional.py

```diff
@@ -1631,3 +1637,274 @@ def rnnt_loss(
+    if isinstance(reference_channel, int):
+        ref_vector = torch.zeros(
+            psd_n.size()[:-3] + psd_n.size()[-1:], device=psd_n.device, dtype=psd_n.dtype
+        )  # (..., channel)
+        ref_vector[..., reference_channel].fill_(1)
+    else:
+        ref_vector = reference_channel
+    # numerator = psd_n.inv() @ stv
+    numerator = torch.linalg.solve(psd_n, rtf).squeeze(-1)  # (..., freq, channel)
+    # denominator = stv^H @ psd_n.inv() @ stv
+    denominator = torch.einsum("...d,...d->...", [rtf.conj().squeeze(-1), numerator])
+    beamform_weights = numerator / (denominator.real.unsqueeze(-1) + eps)
+    # normalzie the numerator
+    if normalize:
+        scale = torch.einsum("...c,...c->...", [rtf.conj().squeeze(-1), ref_vector[..., None, :]])
+        beamform_weights = beamform_weights * scale[..., None]
+
+    return beamform_weights
+
+
+def compute_rtf_evd(psd_s: Tensor) -> Tensor:
+    r"""Estimate the relative transfer function (RTF) or the steering vector by eigenvalue decomposition.
+
+    Args:
+        psd_s (Tensor): The covariance matrix of target speech.
+            Tensor of dimension `(..., freq, channel, channel)`
+
+    Returns:
+        Tensor: The estimated RTF of target speech.
+            Tensor of dimension `(..., freq, channel, 1)`
+    """
+    w, v = torch.linalg.eig(psd_s)  # (..., freq, channel, channel)
+    _, indices = torch.max(w.abs(), dim=-1, keepdim=True)
+    indices = indices.unsqueeze(-1)
+    rtf = v.gather(-1, indices.expand(psd_s.shape[:-1] + (1,)))  # (..., freq, channel, 1)
+    return rtf
+
+
+def compute_rtf_power(psd_s: Tensor, psd_n: Tensor, reference_channel: Union[int, Tensor]) -> Tensor:
+    r"""Estimate the relative transfer function (RTF) or the steering vector by the power method.
+
+    Args:
+        psd_s (Tensor): The covariance matrix of target speech.
+            Tensor of dimension `(..., freq, channel, channel)`
+        psd_n (Tensor): The covariance matrix of noise.
+            Tensor of dimension `(..., freq, channel, channel)`
+        reference_channel (int or Tensor): Indicate the reference channel.
+            If the dtype is ``int``, it represent the reference channel index.
+            If the dtype is ``Tensor``, the dimension is `(..., channel)`.
+
+    Returns:
+        Tensor: the estimated RTF of target speech
+            Tensor of dimension `(..., freq, channel, 1)`
+    """
+    if isinstance(reference_channel, int):
+        ref_vector = torch.zeros(
+            psd_n.size()[:-3] + psd_n.size()[-1:], device=psd_n.device, dtype=psd_n.dtype
+        )  # (..., channel)
+        ref_vector[..., reference_channel].fill_(1)
+    else:
+        ref_vector = reference_channel
+    phi = torch.linalg.solve(psd_n, psd_s)  # psd_n.inv() @ psd_s
+    rtf = torch.einsum("...fec,...c->...fe", [phi, ref_vector])
+    rtf = rtf.unsqueeze(-1)  # (..., freq, channel, 1)
+    rtf = torch.matmul(phi, rtf)
+    rtf = torch.matmul(psd_s, rtf)
```

Comment:
Could you share the codepoint of your implementation? Not sure if it's the pb_bss or other ones.

Reply:
It's not yet published, because I rewrote the code from `torch_complex` to native `torch` and haven't found the time to cleanup, testing and publish the code.

I copied my code to gist.github.com: 
[beamformer.py](https://gist.github.com/boeddeker/ba7226c0f15bc34a610689a47bfeb6f1)

Reply:
In https://arxiv.org/pdf/2011.15003.pdf we wrote a small section about power iteration.
![image](https://user-images.githubusercontent.com/13744128/153432633-08e21e62-a8a0-423c-98f5-6680c4c6328f.png)
When you have only one iteration, you don't need `psd_n`, because it cancels.

In my implementation, I tried to minimize the memory footprint, to avoid an OOM error, because of the backward graph.

Reply:
Got it, `R_n^-1 R_n` becomes `I`. if the iteration is less than 2, you don't even need to compute `\Phi`, instead `PSD_s  matmul u` should suffice. 

Thanks. I will update the code to improve it.

Reply:
Yes. In my implementation, I combined the one hot vector (line 2) with line 4, so I was able to use getitem instead of a matrix multiplication. While this combination is more efficient, it produces less nice code.

Reply:
Addressed in https://github.com/pytorch/audio/pull/2181/commits/07bc84597b3cb9b0b9775634f2bc14d4ddbe3631